instructor/common.py

```diff
@@ -15,18 +15,26 @@
 import json
 import re
 from instructor.compiler import get_protocol_headers
-
+from django.http import HttpResponse
 
 @login_required
 def assignments(request):
     assignments = models.Assignment.objects.filter(course__owner=request.user)
+    return render_to_response(
+        'instructor/assignments.html',
+        {
+            'assignments': assignments
+        },
+        context_instance=RequestContext(request))
 
-    return render_to_response('instructor/assignments.html',
-                              {'assignments': assignments},
-                              context_instance=RequestContext(request))
 @login_required
-def publish_assignment(request, assignment_pk):
+def publish_assignment(request):
+    assignment_pk = request.POST.get('pk')
     assignment = get_object_or_404(models.Assignment, pk=assignment_pk)
+    assignment_ready = is_assignment_complete(assignment)
+    if not assignment_ready:
+        return HttpResponse("Cannot publish unfinished assignment.")
```

Comment:
Can you return an error status code in this case, a 400 would work here. Then you can change your jQuery below to something like

    $.ajax(...).then(function() {
      window.location = ...
    }).fail(function(message) {
      show_alert(message);
    });

In general it's important to use status codes in representing errors so that other libraries can know what is an error and what isn't without parsing your message